source/Octopus.Tentacle/Kubernetes/KubernetesJobMonitorTask.cs

```diff
@@ -0,0 +1,77 @@
+﻿using System;
+using System.Threading;
+using System.Threading.Tasks;
+using Octopus.Diagnostics;
+
+namespace Octopus.Tentacle.Kubernetes
+{
+    public interface IKubernetesJobMonitorTask
+    {
+        void Start();
+        void Stop();
+    }
+
+    public class KubernetesJobMonitorTask: IKubernetesJobMonitorTask, IDisposable
```

Comment:
This structure/pattern is almost a copy of the `WorkspaceCleanerTask` in how it spawns a background thread and manages the lifecycle and cancellation.